nemo/collections/nlp/modules/common/megatron/adapters/mcore_mixins.py

```diff
@@ -63,13 +63,21 @@ def get_query_key_value_tensors(self, hidden_states, key_value_states=None):
         Derives `query`, `key` and `value` tensors from `hidden_states`.
         """
         # Attention heads [sq, b, h] --> [sq, b, ng * (np/ng + 2) * hn)]
-        (mixed_qkv, layernorm_output), _ = self.linear_qkv(hidden_states)
+        linear_qkv_output, _ = self.linear_qkv(hidden_states)
+        layernorm_output = None
+        if isinstance(linear_qkv_output, tuple):  # if LN and linear fused, both will be returned
```

Comment:
Is there an entry in some config for that fusion mentioned here? That should be a more explicit `if` check than `isinstance` of the output.  It would also guide users what happens here so that, for example, they can look up the corresponding option in the config, read docs, etc. The comment is helpful that and is also useful though.

Reply:
Is this only happening with Falcon? Otherwise I'm bit confused how adapters are working for GPT from mcore right now.

Reply:
I only encountered it in Falcon. I think currently GPT default fuses input_layernorm and self_attn. Falcon need to feed input_layernorm to mlp and self_attn at the same time so cannot use the fuse.